compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -5078,3 +5079,76 @@ declare_lint! {
     };
     crate_level_only
 }
+
+declare_lint! {
+    /// The `abi_unsupported_vector_types` lint detects function definitions and calls
+    /// whose ABI depends on enabling certain target features that do not enable those features.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,ignore (fails on non-x86_64)
+    /// #![allow(improper_ctypes_definitions)] // false positive
+    ///
+    /// pub extern "C" fn foo(_: std::arch::x86_64::__m256) {
```

Comment:
Maybe give the function a better name, like `missing_target_feature`? And the other one `with_target_feature`? (or similar)